### DISCUSSION TITLE
Properties of widgets in Java code

### DISCUSSION THREAD

**Discussion by isubsmith** [Gold872/elastic-dashboard#34] [Q&A]
I have a dial widget in a Shuffleboard tab now that goes from 0 to 35 to show our shooter's wrist position.

I'm trying to get a similar widget in Elastic. I can do that if I use a Radial Gauge with angle -90 to 90 and min 0 to 35 from the properties editor. However, I'm not able to do this in the Java code for the robot.

I've tried variations on this:
tuningTab.add("Wrist", 0.0).withWidget(BuiltInWidgets.kDial).withProperties(Map.of("Min", 0, "Max", WristConstants.kWristMaxPosition, "show value", false, "start_angle",-90.0,"end_angle",90.0,"show_pointer", false, "show_ticks", false));

But that shows up as a standard text widget when start Elastic. 

Maybe the first question is - How do I get a Radial Gauge widget instead of the text widget display? 

Thanks!

**Comment by J-Barta** [comment]
I've had a similar issue with the number slider. Same sort of thing - I can eventually get Elastic to work correctly when I play around witht he layout, but it doesn't work when I create the widget in Java.

**Comment by Gold872** [comment]
This is caused by Elastic not mapping the dial widget's name in shuffleboard to the radial gauge widget in Elastic. I'm fixing this by making "Simple Dial" an alias for the radial gauge widget. In the meantime, you can use `.withWidget("Radial Gauge")` and it should create the widget as expected.

> I've had a similar issue with the number slider. Same sort of thing - I can eventually get Elastic to work correctly when I play around witht he layout, but it doesn't work when I create the widget in Java.

Could you describe what exactly was happening? It could be an issue similar to the dial not making the radial gauge. Could I see the code that isn't working the way you're expecting it to?

**Reply by J-Barta** [reply]
I'm sorry, I think I made some mistake at some point. I just went to reproduce the issue by forcing Elastic to regenerate the layout based on my Java code, and it worked correctly this time. The previous behavior I observed might have been caused bysome sort of network tables shenanigan or and older version of Elastic.

**Comment by isubsmith** [comment]
> This is caused by Elastic not mapping the dial widget's name in shuffleboard to the radial gauge widget in Elastic. I'm fixing this by making "Simple Dial" an alias for the radial gauge widget. In the meantime, you can use `.withWidget("Radial Gauge")` and it should create the widget as expected.

That worked, thanks. Id would be nice to be able to switch between Shuffleboard and Elastic trivially, but I suspect we'll switch to Elastic fully so a minor thing.

One more question - how do I actually build Elastic? I have a minor change to NumberSlider that I think should allow for a vertical orientation (based on code for NumberBar). I just submitted this patch to Shuffleboard last week :)

**Reply by Gold872** [reply]
I'm going to assume you're on Windows, but Flutter has documentation for developing on all desktop platforms. There's instructions for installing here: https://docs.flutter.dev/get-started/install/windows/desktop

Once you have the Flutter SDK installed, clone the repository, and run `flutter run` in the directory you cloned it into

Let me know if you run into any issues, the initial setup on Windows can be a bit convoluted
